app/uk/gov/hmrc/vatregisteredcompanies/repositories/VatRegisteredCompaniesRepository.scala

```diff
@@ -58,39 +69,48 @@ object Wrapper {
 
 @Singleton
 class   VatRegisteredCompaniesRepository @Inject()(
-  reactiveMongoComponent: ReactiveMongoComponent,
+  mongoComponent: MongoComponent,
   bufferRepository: PayloadBufferRepository,
   @Named("deletionThrottleElements") elements: Int,
   @Named("deletionThrottlePer") per: FiniteDuration
 )(implicit val executionContext: ExecutionContext, mat: Materializer) extends
-  ReactiveRepository("vatregisteredcompanies", reactiveMongoComponent.mongoConnector.db, Wrapper.format) {
+  PlayMongoRepository[Wrapper](
+    mongoComponent = mongoComponent,
+    collectionName = "vatregisteredcompanies",
+    domainFormat = Wrapper.format,
+    indexes = Seq(IndexModel(ascending("vatNumber"),
+      IndexOptions().name("vatNumberIndexNew").unique(false).background(true)))) with Logging {
+
 
   def deleteAll(): Future[Unit] =
-    removeAll().map(_=> ())
+    collection.deleteMany(Filters.empty()).toFuture().map(_ => ())
 
-  private val im: CollectionIndexesManager = collection.indexesManager
+  //private val im: CollectionIndexesManager = collection.indexesManager
```

Comment:
remove the commented code

Reply:
the commented code was left for you to see how we made our changes, let me know if you want me to delete them and push up again

Reply:
removed